gobblin-api/src/main/java/gobblin/writer/Destination.java

```diff
@@ -27,6 +27,7 @@
    */
   public static enum DestinationType {
     HDFS,
+    HDFS_TIME_PARTITION,
```

Comment:
It does not make sense to add a new destination as DestinationType is used to indicate the target data system the output data goes to. In this case, data is still written to HDFS, but just using a different folder structure. Instead of adding a new destination and a new switch branch in AvroDataWriterBuilder, you should create a new DataWriterBuilder especially since you have a new writer class.